customization/patch_image_registires.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd ) ; readonly dir
+cd "${dir}/.."
+
+
+# -----------------------------------------------------------------------------
+# {{- define "cilium.image" -}} and {{- define "cilium.operator.image" -}}
+# patches adding .Values.image.registry to all images.
+# -----------------------------------------------------------------------------
+
+#
+# This is the final definition we want:
+#
+#    {{- define "cilium.image" -}}
+#    {{- if not (kindIs "slice" .) }}
+#    {{- (fail (printf "required list, but got %q" (kindOf .))) }}
+#    {{- end }}
+#    {{- if (ne (len .) 2) }}
+#    {{- (fail (printf "required list of 2 arguments, but got %d" (len .))) }}
+#    {{- end }}
+#    {{- $ := index . 0 }}
+#    {{- with index . 1 }}
+#    {{- $digest := (.useDigest | default false) | ternary (printf "@%s" .digest) "" -}}
+#    {{- if .override -}}
+#    {{- printf "%s" .override -}}
+#    {{- else -}}
+#    {{- printf "%s/%s:%s%s" $.Values.image.registry .repository .tag $digest -}}
+#    {{- end -}}
+#    {{- end -}}
+#    {{- end -}}
+# 
+# This needs to be applied to 2 files:
+#
+#   - helm/cilium/templates/_helpers.tpl
+#   - helm/cilium/templates/cilium-operator/_helpers.tpl
+#
+
+cat <<'EOF' > /tmp/patch.txt
```

Comment:
I don't think it's a good idea to keep the patches embedded in this script. Instead I'd have something like:

- `patches/00-image-registry.diff`
- `patches/01-another-small-change.diff`
- `patches/02-some-other-changes.diff`

And this script to go over them in order. This would help with the "creating patches" workflow.